electrum/plugins/bitbox02/bitbox02.py

```diff
@@ -0,0 +1,628 @@
+            privkey = bitbox02_config.get("noise_privkey")
+            if privkey is not None:
+                return bytes.fromhex(privkey)
+            return None
+
+        def set_noise_privkey(privkey: bytes) -> None:
+            bitbox02_config = self.config.get("bitbox02")
+            bitbox02_config["noise_privkey"] = privkey.hex()
+            self.config.set_key("bitbox02", bitbox02_config)
+
+        def attestation_warning() -> None:
+            self.handler.attestation_failed_warning(
+                "The BitBox02 attestation failed.\nTry reconnecting the BitBox02.\nWarning: The device might not be genuine, if the\n problem persists please contact Shift support."
+            )
+
+        class NoiseConfig(bitbox_api_protocol.BitBoxNoiseConfig):
+            """NoiseConfig extends BitBoxNoiseConfig"""
+
+            def show_pairing(self, code: str) -> bool:
+                choice = [code]
+                try:
+                    reply = pairing_step(code)
+                except:
+                    # Close the hid device on exception
+                    hid_device.close()
+                    raise
+                return True
+
+            def attestation_check(self, result: bool) -> None:
+                if not result:
+                    attestation_warning()
+
+            def contains_device_static_pubkey(self, pubkey: bytes) -> bool:
+                return exists_remote_static_pubkey(pubkey)
+
+            def add_device_static_pubkey(self, pubkey: bytes) -> None:
+                return set_remote_static_pubkey(pubkey)
+
+            def get_app_static_privkey(self) -> Optional[bytes]:
+                return get_noise_privkey()
+
+            def set_app_static_privkey(self, privkey: bytes) -> None:
+                return set_noise_privkey(privkey)
+
+        if self.bitbox02_device is None:
+            hid_device = hid.device()
+            hid_device.open_path(self.bitbox_hid_info["path"])
+
+            self.bitbox02_device = bitbox02.BitBox02(
+                transport=u2fhid.U2FHid(hid_device),
+                device_info=self.bitbox_hid_info,
+                noise_config=NoiseConfig(),
+            )
+
+        if not self.bitbox02_device.device_info()["initialized"]:
+            raise BaseException(
+                "Please initialize the BitBox02 using the BitBox app first before using the BitBox02 in electrum"
+            )
+
+    def check_device_firmware_version(self) -> bool:
+        if self.bitbox02_device is None:
+            raise BaseException(
+                "Need to setup communication first before attempting any BitBox02 calls"
+            )
+        return self.bitbox02_device.check_firmware_version()
+
+    def coin_network_from_bip32_list(self, keypath: List[int]) -> int:
+        if len(keypath) > 2:
+            if keypath[1] == 1 + HARDENED:
+                return bitbox02.btc.TBTC
+        return bitbox02.btc.BTC
```

Comment:
This is a bit weird... Is it a conscious decision that you are choosing testnet magic bytes based on the bip32 path, and not e.g. `constants.net.TESTNET` (which is set for the process if we were started in testnet mode)?

Reply:
It's a hack to get the testnet xpub for the wallet encryption. Once the encryption is fixed, I'd move to the constant as well.